### ISSUE TITLE
Static deltas are broken with latest ostree

### ISSUE THREAD

**Issue by vnd** [ostreedev/ostree#2680]
I'm trying to benchmark static deltas and estimate bandwidth usage, compared to 'archive' ostree updates.
However reported update size with static deltas seems to be roughly the same as without. I'm probably doing something wrong, but it all looks pretty straightforward, so not sure what's wrong.

Procedure, server side:
```
ostree --repo=repo --mode=archive init
ostree --repo=repo commit --branch=v1 --subject="Build 1" rootfs
# copy slightly newer version of rootfs to the 'rootfs' folder, with a bunch of binary files being different
ostree --repo=repo commit --branch=v1 --subject="Build 2" rootfs
ostree --repo=repo static-delta generate v1 # optional step
cd repo && python3 -m http.server 8000
```
Procedure, client side (same machine):
```
ostree --repo=repo init
ostree --repo=repo remote add --no-gpg-verify local http://localhost:8000 --repo=repo
ostree --repo=repo pull local v1
```
ostree pull is done twice: after the first build is committed to the server, and then after the second build is committed.
Second pull, with static deltas:
`1 delta parts, 68 loose fetched; 4702 KiB transferred in 0 seconds; 1,1 MB content written`
Without static deltas:
`19 metadata, 75 content objects fetched; 4912 KiB transferred in 0 seconds; 13,4 MB content written`

So static deltas are fetched, but for some reason transferred size is roughly the same.

Thanks!

**Comment by vnd** [comment]
One suspect was incremented minor .so versions -- I thought that may be they are treated as new files and fully downloaded. So I've limited rootfs changes only to some of /usr/bin/* files, each of those produces bsdiff of only about 200 bytes (!) when doing it manually. Still, what I got from ostree pull is:
`1 delta parts, 26 loose fetched; 3197 KiB transferred in 0 seconds; 4,1 MB content written`

**Comment by vnd** [comment]
Ok, so this looks like a (major?) regression in ostree, since 2020.3 works perfectly (160 KB with static deltas).

Attached are client and server logs of the procedure, hope someone will need them.

[client.log](https://github.com/ostreedev/ostree/files/9259940/client.log)
[server.log](https://github.com/ostreedev/ostree/files/9259943/server.log)

**Comment by dbnicholson** [comment]
Can you run `ostree static-delta show` on the generated delta?

**Comment by vnd** [comment]
```
$ ostree --repo=repo static-delta list
9511a56922f23822aed11e12db9a30969d03fee680fc985c6ad2be364d92f90c-886b8d6ceae0fa8365919f70e697c482df0eee4640863a0e0489caa3ddbceec7
t480s ~/Downloads/builds/alio-vend 
$ ostree --repo=repo static-delta show 9511a56922f23822aed11e12db9a30969d03fee680fc985c6ad2be364d92f90c-886b8d6ceae0fa8365919f70e697c482df0eee4640863a0e0489caa3ddbceec7
Delta: 9511a56922f23822aed11e12db9a30969d03fee680fc985c6ad2be364d92f90c-886b8d6ceae0fa8365919f70e697c482df0eee4640863a0e0489caa3ddbceec7
Signed: no
From: 9511a56922f23822aed11e12db9a30969d03fee680fc985c6ad2be364d92f90c
To: 886b8d6ceae0fa8365919f70e697c482df0eee4640863a0e0489caa3ddbceec7
Endianness: little
Timestamp: 1659679857
Number of parents: 0
Number of fallback entries: 0
Total Fallback Size: 0 (0 bytes)
Total Fallback Uncompressed Size: 0 (0 bytes)
Number of parts: 1
PartMeta0: nobjects=2 size=1049605 usize=4194348
PartPayload0: nmodes=1 nxattrs=1 blobsize=4194404 opsize=20
PartPayloadOps0: openspliceclose=1 open=1 write=0 setread=1 unsetread=1 close=1 bspatch=1
Total Part Size: 1049605 (1,0 MB)
Total Part Uncompressed Size: 4194348 (4,2 MB)
Total Size: 1049605 (1,0 MB)
Total Uncompressed Size: 4194348 (4,2 MB)
```
Note: I've repeated the scenario, so hashes are different from client.log and server.log

**Comment by dbnicholson** [comment]
Also, what version of ostree is this?

**Comment by vnd** [comment]
It's latest ostree from master, and also 2022.5 from Arch Linux. Log is attached.
[client-verbose.log](https://github.com/ostreedev/ostree/files/9269276/client-verbose.log)

**Comment by vnd** [comment]
Eventually fixed by #2054 , thanks @jlebon
